qhub/cli/_init.py

```diff
@@ -0,0 +1,496 @@
+            )
+        )
+
+        os.environ["GITHUB_CLIENT_ID"] = typer.prompt(
+            "Please enter your GITHUB_CLIENT_ID",
+            hide_input=True,
+        )
+        os.environ["GITHUB_CLIENT_SECRET"] = typer.prompt(
+            "Please enter your GITHUB_CLIENT_SECRET",
+            hide_input=True,
+        )
+
+    return auth_provider
+
+
+def check_project_name(ctx: typer.Context, project_name: str):
+    """Validate the project_name is acceptable. Depends on `cloud_provider`."""
+
+    project_name_convention(
+        project_name.lower(), {"provider": ctx.params["cloud_provider"]}
+    )
+
+    return project_name
+
+
+def guided_init_wizard(ctx: typer.Context, guided_init: str):
+    """
+    Guided Init Wizard is a user-friendly questionnaire used to help generate the `nebari-config.yaml`.
+    """
+    qmark = "  "
+    disable_checks = os.environ.get("QHUB_DISABLE_INIT_CHECKS", False)
+
+    if not guided_init:
+        return guided_init
+
+    rich.print(
+        (
+            "\n\t\t[bold]Welcome to the Guided Init wizard![/bold]\n\n"
+            "You will be asked a few questions that are used to generate your [purple]nebari-config.yaml[/purple]. "
+            f"{LINKS_TO_DOCS_TEMPLATE.format(link_to_docs=DOCS_HOME)}"
+        )
+    )
+
+    if disable_checks:
+        rich.print(
+            "⚠️  Attempting to use the Guided Init wizard without any validation checks. There is no guarantee values provided will work!  ⚠️\n\n"
+        )
+
+    # pull in default values for each of the below
+    inputs = InitInputs()
+
+    # CLOUD PROVIDER
+    rich.print(
+        (
+            "\n 🪴  Nebari runs on a Kubernetes cluster so one of the first choices that needs to be made "
+            "is where you want this Kubernetes cluster deployed. "
+            f"{LINKS_TO_DOCS_TEMPLATE.format(link_to_docs=CHOOSE_CLOUD_PROVIDER)}"
+            "\n\t❕ [purple]local[/purple] requires Docker and Kubernetes running on your local machine. "
+            "[italic]Currently only available on Linux OS.[/italic]"
+            "\n\t❕ [purple]existing[/purple] refers to an existing Kubernetes cluster that Nebari can be deployed on.\n"
+        )
+    )
+    inputs.cloud_provider = questionary.select(
+        "Where would you like to deploy your Nebari cluster?",
+        choices=enum_to_list(ProviderEnum),
+        qmark=qmark,
+    ).ask()
+
+    if not disable_checks:
+        check_cloud_provider_creds(ctx, cloud_provider=inputs.cloud_provider)
+
+    # specific context needed when `check_project_name` is called
+    ctx.params["cloud_provider"] = inputs.cloud_provider
+
+    # PROJECT NAME
+    rich.print(
+        (
+            "\n 🪴  Next, give your Nebari instance a project name. This name is what your Kubernetes cluster will be referred to as.\n\n"
+        )
+    )
+    inputs.project_name = questionary.text(
+        "What project name would you like to use?",
```

Comment:
I got the following `ValueError` the first time I ran this, which caused me to exit the guided init wizard.

```python-traceback
 ValueError: 
    In order to successfully deploy QHub, there are some project naming conventions which need
    to be followed. First, ensure your name is compatible with the specific one for
    your chosen Cloud provider. In addition, the QHub project name should also obey the following
    format requirements:
    - Letters from A to Z (upper and lower case) and numbers;
    - Maximum accepted length of the name string is 16 characters.
    - If using AWS: names should not start with the string "aws";
    - If using Azure: names should not contain "-".

 Maximum accepted length of the project name string is 16 characters.
```

A couple of suggestions:
- We can share these naming guidelines in the question itself.
- If possible, it'll be nice to swallow the Error and not close the wizard entirely. Instead, we can display a new question asking users to retry the name.

Reply:
Added :)